Add Feedstock repository structure section (#1786)
conda-forge/conda-forge.github.io: src/maintainer/adding_pkgs.rst

* If you want to make a change to the recipe, send a :term:`PR` to the git repository from a fork. Branches of the main repository are used for maintaining different versions only.

.. _feedstock-repository-structure:

Feedstock repository structure
------------------------------

> Once the PR having the recipe for a package is merged in the ``staged-recipes`` repository, a new repository is created automatically called ``<package-name>-feedstock``.

Reviewer: I feel like "having" isn't the best word here. Maybe "containing" or just "for", though then we could have two instances of the word "for" pretty close together. Up to you.

> A feedstock is made up of a conda recipe (the instructions on what and how to build the package) and the necessary configuration files for automatic builds using freely available continuous integration services.

Reviewer: Let's add "(CI)" after "continuous integration", since we mention CI configuration files later in this section. That will make things more clear.

> Each feedstock contains various files which are generated automatically using our automated provisioning tool `conda-smithy <https://github.com/conda-forge/conda-smithy/>`__. Broadly every feedstock has the following files:

Reviewer: ```suggestion
Each feedstock contains various files that are generated automatically using our automated provisioning tool `conda-smithy <https://github.com/conda-forge/conda-smithy/>`__. Broadly, every feedstock has the following files:
```

recipe
......

This folder contains the ``meta.yaml`` file and any other files/scripts needed to build the package.

LICENSE.txt
............

This file is the license for the recipe itself. This license is different from the package license, which you define while submitting the package recipe using ``license_file`` in the ``meta.yaml`` file.

CI-files
........

These are the CI configuration files for service providers like Azure and TravisCI.

conda-forge.yml
................

> This file is used to configure how the feedstock is set up and built. Making any changes in this file usually requires `rerending of the feedstock <https://conda-forge.org/docs/maintainer/updating_pkgs.html#dev-update-rerender>`__.

Reviewer: ```suggestion
This file is used to configure how the feedstock is set up and built. Making any changes in this file usually requires `rerendering the feedstock <https://conda-forge.org/docs/maintainer/updating_pkgs.html#dev-update-rerender>`__.
```

* If you want to make a change to the recipe, send a :term:`PR` to the git repository from a fork. Branches of the main repository are used for maintaining different versions only.

.. _feedstock-repository-structure:

Feedstock repository structure
------------------------------

> Once the PR having the recipe for a package is merged in the ``staged-recipes`` repository, a new repository is created automatically called ``<package-name> feedstock`` which lies in the `feedstock <https://github.com/conda-forge/feedstocks/>`__ repository.

Reviewer 2: ```suggestion
Once the PR having the recipe for a package is merged in the ``staged-recipes`` repository, a new repository is created automatically called ``<package-name>-feedstock``.
```

> A feedstock is made up of a conda recipe (the instructions on what and how to build the package) and the necessary configuration files for automatic building using freely available continuous integration services.

Reviewer 2: ```suggestion
A feedstock is made up of a conda recipe (the instructions on what and how to build the package) and the necessary configuration files for automatic builds using freely available continuous integration services.
```

> Each feedstock contains various files which are generated automatically using `conda-smithy <https://github.com/conda-forge/conda-smithy/>`__ which is a tool used for managing continuous integration and maintenance of the feedstocks. Broadly every feedstock has the following files : 

Reviewer 2: ```suggestion
Each feedstock contains various files which are generated automatically using our automated provisioning tool `conda-smithy <https://github.com/conda-forge/conda-smithy/>`__. Broadly every feedstock has the following files: 
```

recipe
......

> This folder contians the ``meta.yaml`` file in which everything is defined that is required to build and use the package.

Reviewer 2: ```suggestion
This folder contains the ``meta.yaml`` file and any other files/scripts needed to build the package.
```

LICENSE.txt
............

> This is an auto-generated license which is generated when feedstock is created. This license is different from the related package license, which you define while submitting the package recipe using ``license_file`` in the ``meta.yaml`` file.

Reviewer 2: ```suggestion
This file is the license for the recipe itself. This license is different from the package license, which you define while submitting the package recipe using ``license_file`` in the ``meta.yaml`` file.
```

CI-files
........

> These are the CI configuration files for service providers like CircleCI, AppVeyor and TravisCI, which makes it possible to build and upload installable packages to the conda-forge Anaconda-Cloud channel for Linux, Windows and OSX respectively.

Reviewer 2: ```suggestion
These are the CI configuration files for service providers like Azure and TravisCI.
```

> ``.scripts``
> .............
> 
> This folder contains build scripts for various environments.

Reviewer 2: ```suggestion
```

conda-forge.yml
................

> This file is used to configure how conda-forge is set up and built. Making any changes in this files usually requires `rerending of the feedstock <https://conda-forge.org/docs/maintainer/updating_pkgs.html#dev-update-rerender>`__.

Reviewer 2: ```suggestion
This file is used to configure how the feedstock is set up and built. Making any changes in this file usually requires `rerending of the feedstock <https://conda-forge.org/docs/maintainer/updating_pkgs.html#dev-update-rerender>`__.
```

conda-forge.yml
................

> This file is used to configure how the feedstock is set up and built. Making any changes in this file usually requires `rerending the feedstock <https://conda-forge.org/docs/maintainer/updating_pkgs.html#dev-update-rerender>`__.

Reviewer: Should be "rerendering", not "rerending" in that link text.
